### ISSUE TITLE
X axes not synched

### ISSUE THREAD

**Issue by rabergj** [KopfLab/ggstackplot#17]
Seems that a recent fix (remove_na? colors?) generated another issue - the x axes are no longer synched.

```{r}
synth_data <- 
  mtcars |>
  cross_join(
    tibble(
      calib = factor(c("a", "b", "c")),
      diff = c(-0.8, 1, 0.8)
    )
  ) |>
  mutate(
    across(c(wt, qsec, drat), ~.x * diff),
    wt = ifelse(calib == "b", wt, NA),
    drat = ifelse(calib != "a", drat, NA)
  ) |> 
  #  ***changing the x axis for a subset of the data***
  mutate(mpg = ifelse(calib == "a", mpg * 2, mpg))
  
synth_data |>
  ggstackplot(
    x = mpg,
    y = c(wt, qsec, drat),
    template = ggplot() +
      aes(color = calib) +
      geom_line() +
      # the drop = FALSE is key, otherwise you'll loose color matching
      scale_color_brewer(palette = "Set1", drop = FALSE) +
      theme_stackplot()
  )
```

![image](https://github.com/KopfLab/ggstackplot/assets/35500268/dafc7d98-d50a-452c-8e26-f6e7605b6204)

**Comment by sebkopf** [comment]
@rabergj they actually are, try it with `both_axes = TRUE` - the problem is that the legends are not lined up (don't think that's possible) and since the axis ticks and labels have different widths, the legends get pushed out. This leads back to issue #10 - how do we deal with legends?

Close this one though?

**Comment by sebkopf** [comment]
oh i see what you mean! yes, the x axis at present assume identical range, i'll make sure to fix that

**Comment by sebkopf** [comment]
btw @rabergj the way to really show this problem is by turning `simplify_shared_axis = FALSE` to get all x-axes displayed (btw happy for input about all argument names too, if they don't make sense to any of you, they won't make sense to other users either)

**Comment by sebkopf** [comment]
question @rabergj  implement common shared axis range (i.e. range synched x axis) only if `simplify_shared_axis = TRUE` or always? (is there a point in having stacked plots with unsynched x axis or should folks just use a facet wrap at that point?)

**Comment by sebkopf** [comment]
@rabergj this should be fixed in version 0.2.6, please test and reopen if not
